Sources/Core/Shared/Operation.swift

```diff
@@ -76,6 +76,24 @@ public class Operation: NSOperation {
         }
     }
 
+    /** 
+     Type to express the intent of the user in regards to executing an Operation instance
+     
+     - see: https://developer.apple.com/library/ios/documentation/Performance/Conceptual/EnergyGuide-iOS/PrioritizeWorkWithQoS.html#//apple_ref/doc/uid/TP40015243-CH39
+    */
+    public enum UserIntent: Int {
+        case Background = 0, SideEffect, Initiated
+
+        internal var qos: NSQualityOfService {
+            switch self {
+            case .Initiated, .SideEffect:
+                return .UserInitiated
+            default:
+                return .Default
+            }
```

Comment:
I wonder if `.Background` is misleading since that doesn't actually correlate to a `.Background` QoS (and shouldn't since that would just be duplicating QoS). I wonder if `None`, `Implicit`, `Indirect`, or `Unintended` might be better. My vote would be for `None`, as in "the user had no intention about this operation."

Also, what is the intention behind `SideEffect`?